ldap/servers/slapd/slap.h

```diff
@@ -279,8 +279,7 @@ typedef void (*VFPV)(); /* takes undefined arguments */
 #define SLAPD_DEFAULT_MAXBERSIZE_STR "2097152"
 #define SLAPD_DEFAULT_MAXSIMPLEPAGED_PER_CONN (-1)
 #define SLAPD_DEFAULT_MAXSIMPLEPAGED_PER_CONN_STR "-1"
-/* We'd like this number to be prime for the hash into the Connection table */
-#define SLAPD_DEFAULT_CONNTABLESIZE 4093 /* connection table size */
+#define SLAPD_DEFAULT_CONNTABLESIZE 65536 /* connection table size */
```

Comment:
This seems like a really large number - remind me, is this conntable size bound by max fd's somewhere? Else we'll hit a different set of resource exhaustion issues.

Reply:
Hi @Firstyear 
I had a chat with the team and it seems a connection table size of 65k is not an uncommon deployment, so we went with that. 

Regarding resource limits, as long as the soft limit is greater than conntablesize then we are good, during startup. after the config is loaded the server increases the soft limit to match the hard limit.

Reply:
I did suggest, but I think it was overlooked, that we can dynamically set the conntable size, like how we set the FD limit in DS.  We do it start up, and it can be seen in the errors log.

Reply:
@mreynolds389 
Apologies if you think I ignored your suggestion, that wasn't my intention. 
Currently the conntablesize is set dynamically to the system fd soft limit, by default this is 1024 and the root of our problem. From what I understand the fd limits in DS are based on the systems soft and hard fd limits so I don't see how we can make the conntablesize dynamic, unless we do something to calculate what we think it should be.

Reply:
There are no such thing as root causes, only contributing factors :) 

A dynamic conntable would be a nightmare given the need to realloc the table. I think it would be dynamic with regard to restarts and reading the fd limit and choosing a %of that fd limit to set the conntable amount. But that has other risks I think if the number is too high.

Reply:
We could have a self tuning conntablesize, where the server initially starts up with a default value, gets a count of the maximum number of connections the server had to handle, on shutdown/restart the server sets the conntablesize attribute to the max number of connections handled. Or is this overkill ?

Reply:
Hi @mreynolds389 , 
 I am not sure what you mean exactly:
 As I understand it, the idea behind config_set_maxdescriptors is to the the fi;e descriptor limit according to the dse.ldif value (and decrease the configured value if the system cannot use that value) 
 But here we want to compute the connection default value (BTW: in the libglob,c code that is removed
 in this PR, The config parameter are not yet even read) 
  Now I do not think that we could set  the conn table size to the config_set_maxdescriptors
    value (because there is a serious risk of db corruption if we run out of fds) 
But IMHO the proposed code is buggy and will leads to trouble if config_set_maxdescriptors set the value below 64K ( I think that we need to check before creating the conntectiom table  that 
  maxconn + reserveddesc <= maxdesc and decrease maxconn if it is not the case 
    and log an info message in that case )

Reply:
@progier this what I was referring to

```
    if (getrlimit(RLIMIT_NOFILE, &rlp) == 0) {
            conntable =  (int64_t)rlp.rlim_max - 64  // something like this
    }
```

Reply:
Hum ! would rather use config_get_reservedescriptors() instead of 64 
(Using an hardcoded value of 64 seems dangerous as some config have more than 64 indexes) 
and I would also add a second test to limit the value to 64K 
But I agree with the idea ( and we should also fix the doc because the nsslapd-conntablesize value will then be ignored.)
I would also add a test to increase the reserveddescriptor value if it is below 64 + 3*nbBackends + nbIndexes

Reply:
> Hum ! would rather use config_get_reservedescriptors() instead of 64 (Using an hardcoded value of 64 seems dangerous as some config have more than 64 indexes) 

Sorry again it was a rough example, like I said "something like this"  :-)  I said 64 because that is what that reserve descriptors are, but I was just trying to highlight my approach since I've been previously misunderstood.  

> and I would also add a second test to limit the value to 64K But I agree with the idea ( and we should also fix the doc because the nsslapd-conntablesize value will then be ignored.) I would also add a test to increase the reserveddescriptor value if it is below 64 + 3*nbBackends + nbIndexes

+1

Reply:
This commit removes a configurable conntablesize, we now calculate it using (maxdesc - reservedesc), A check has been added for the process 64k soft limit. 

Regarding the reserveddescriptor value, I noticed that on my dev vm the non client connection fds used by the server are  greater than 64. So how about we have a dynamic reserve descriptors  value using the formula from the RH DS docs.
![image](https://user-images.githubusercontent.com/71395320/196453956-a899be5e-bafe-4f82-8af9-ead8042f42df.png)

Reply:
Not sure we should  make reserve descriptor fully dynamic (as unknown plugins may need file descriptors)
IMHO we should dynamically increase the configured value if the formula shows it is too small 
otherwise I am ok with your formula ...

Reply:
> Not sure we should make reserve descriptor fully dynamic (as unknown plugins may need file descriptors)
> IMHO we should dynamically increase the configured value if the formula shows it is too small
> otherwise I am ok with your formula ...

Yes this makes sense, I never considered future plugins that might require fds. I will  create a separate PR for the reserve-descriptor change.